lua/telescope/builtin/files.lua

```diff
@@ -187,30 +188,42 @@ files.current_buffer_fuzzy_find = function(opts)
 end
 
 files.tags = function(opts)
-  local ctags_file = opts.ctags_file or 'tags'
+  local tagfiles = opts.ctags_file and { opts.ctags_file } or
+                   vim.fn.map(vim.fn.tagfiles(), 'fnamemodify(v:val, ":p")')
 
-  if not vim.loop.fs_open(vim.fn.expand(ctags_file), "r", 438) then
-    print('Tags file does not exists. Create one with ctags -R')
+  if vim.tbl_isempty(tagfiles) then
+    print('No tags file found. Create one with ctags -R or verify your &tags option is correctly set.')
     return
   end
 
-  local fd = assert(vim.loop.fs_open(vim.fn.expand(ctags_file), "r", 438))
-  local stat = assert(vim.loop.fs_fstat(fd))
-  local data = assert(vim.loop.fs_read(fd, stat.size, 0))
-  assert(vim.loop.fs_close(fd))
+  opts.entry_maker = opts.entry_maker or make_entry.gen_from_ctags(opts)
+  local finder = finders.new_incremental(opts)
 
-  local results = vim.split(data, '\n')
+  for _, each in ipairs(tagfiles) do
+    local tags_directory = vim.fn.fnamemodify(each, ":h")
+    local resolve_filename = function(filename)
+        return path.exists(filename) and filename or
+               path.join(tags_directory, filename)
+    end
+    path.read(each, function(err, data)
+      for _, line in ipairs(vim.split(data, '\n')) do
+        finder:feed({ line = line, resolve_filename = resolve_filename })
+      end
+    end)
+    finder:finish()
```

Comment:
I also think this `finish` is placed at the wrong position. We don't want to `finish` after the first `tagfile` is processed. We want files be loaded

Reply:
Thanks, I had in mind to do that (a counter) but forgot to actually implement it -- there now I think.